nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutTCP.java

```diff
@@ -168,61 +144,23 @@ public void onTrigger(final ProcessContext context, final ProcessSessionFactory
         final ProcessSession session = sessionFactory.createSession();
         final FlowFile flowFile = session.get();
         if (flowFile == null) {
-            final PruneResult result = pruneIdleSenders(context.getProperty(IDLE_EXPIRATION).asTimePeriod(TimeUnit.MILLISECONDS).longValue());
-            // yield if we closed an idle connection, or if there were no connections in the first place
-            if (result.getNumClosed() > 0 || (result.getNumClosed() == 0 && result.getNumConsidered() == 0)) {
-                context.yield();
-            }
             return;
         }
 
-        ChannelSender sender = acquireSender(context, session, flowFile);
-        if (sender == null) {
+        if (eventSender == null) {
             return;
         }
 
-        // really shouldn't happen since we know the protocol is TCP here, but this is more graceful so we
-        // can cast to a SocketChannelSender later in order to obtain the OutputStream
-        if (!(sender instanceof SocketChannelSender)) {
-            getLogger().error("Processor can only be used with a SocketChannelSender, but obtained: " + sender.getClass().getCanonicalName());
-            context.yield();
-            return;
-        }
-
-        boolean closeSender = isConnectionPerFlowFile(context);
         try {
-            // We might keep the connection open across invocations of the processor so don't auto-close this
-            final OutputStream out = ((SocketChannelSender)sender).getOutputStream();
-            final String delimiter = getOutgoingMessageDelimiter(context, flowFile);
-
-            final StopWatch stopWatch = new StopWatch(true);
-            try (final InputStream rawIn = session.read(flowFile);
-                 final BufferedInputStream in = new BufferedInputStream(rawIn)) {
-                IOUtils.copy(in, out);
-                if (delimiter != null) {
-                    final Charset charSet = Charset.forName(context.getProperty(CHARSET).getValue());
-                    out.write(delimiter.getBytes(charSet), 0, delimiter.length());
-                }
-                out.flush();
-            } catch (final Exception e) {
-                closeSender = true;
-                throw e;
-            }
-
+            byte[] content = readContent(context, session, flowFile);
```

Comment:
Recommend adding `final`:
```suggestion
            final byte[] content = readContent(context, session, flowFile);
```